anitya/lib/backends/crates.py

```diff
@@ -0,0 +1,216 @@
+
+    name = 'crates.io'
+    examples = [
+        'https://crates.io/crates/clap',
+        'https://crates.io/crates/serde',
+    ]
+
+    @classmethod
+    def _get_versions(cls, project):
+        """
+        Make a request for all versions of the project provided.
+
+        Args:
+            project (anitya.lib.model.Project): The Rust project to retrieve
+                versions for.
+
+        Returns:
+            list: A list of version dictionaries. Each dictionary is in the format::
+                {
+                    "crate": "itoa",
+                    "created_at": "2017-01-26T05:02:29Z",
+                    "dl_path": "/api/v1/crates/itoa/0.2.1/download",
+                    "downloads": 150,
+                    "features": {},
+                    "id": 43354,
+                    "links": {
+                        "authors": "/api/v1/crates/itoa/0.2.1/authors",
+                        "dependencies": "/api/v1/crates/itoa/0.2.1/dependencies",
+                        "version_downloads": "/api/v1/crates/itoa/0.2.1/downloads"
+                    },
+                    "num": "0.2.1",
+                    "updated_at": "2017-01-26T05:02:29Z",
+                    "yanked": false
+                }
+
+        Raises:
+            AnityaPluginException: If the URL was unreachable or the response
+                was in an unexpected format.
+        """
+        url = 'https://crates.io/api/v1/crates/{}/versions'.format(project.name)
+        try:
+            req = cls.call_url(url)
+            req.raise_for_status()
+            data = req.json()
+        except requests.RequestException as e:
+            raise AnityaPluginException('Could not contact {url}: '
+                                        '{reason!r}'.format(url=url, reason=e))
+        except ValueError as e:
+            raise AnityaPluginException('Failed to decode JSON: {!r}'.format(e))
+
+        return data['versions']
+
+    @classmethod
+    def get_version(cls, project):
+        """
+        Get the latest version of the project provided.
+
+        Args:
+            project (anitya.lib.model.Project): The Rust project to retrieve
+                the latest version for.
+
+        Returns:
+            str: A version string.
+
+        Raises:
+            AnityaPluginException: If the URL was unreachable or the response
+                was in an unexpected format.
+        """
+        return cls._get_versions(project)[0]['num']
+
+    @classmethod
+    def get_versions(cls, project):
+        """
+        Get all versions of the project provided.
+
+        Args:
+            project (anitya.lib.model.Project): The Rust project to retrieve
+                versions for.
+
+        Returns:
+            list: A list of version strings.
+
+        Raises:
+            AnityaPluginException: If the URL was unreachable or the response
+                was in an unexpected format.
+        """
```

Comment:
Fair point. I added a section to the contribution guide about documentation and mention the style used (https://github.com/fedora-infra/anitya/pull/421). I know all the existing documentation uses the plain RST style, but I find this style much easier to read inline and I plan to convert any old strings as I go along.

They'll both render fine in HTML so it's just a matter of consistency in the code.